dnf/base.py

```diff
@@ -2500,6 +2498,29 @@ def _report_already_installed(self, packages):
         for pkg in packages:
             _msg_installed(pkg)
 
+    def _raise_package_not_found_error(self, pkg_spec, forms, reponame):
+        all_query = self.sack.query(flags=hawkey.IGNORE_EXCLUDES)
+        subject = dnf.subject.Subject(pkg_spec)
+        solution = subject.get_best_solution(
+            self.sack, forms=forms, with_src=False, query=all_query)
+        if reponame is not None:
+            solution['query'].filterm(reponame=reponame)
+        if not solution['query']:
+            raise dnf.exceptions.PackageNotFoundError(_('No match for argument'), pkg_spec)
+        else:
+            with_modular_query = self.sack.query(flags=hawkey.IGNORE_MODULAR_EXCLUDES)
+            with_modular_query = solution['query'].intersection(with_modular_query)
+            with_regular_query = self.sack.query(flags=hawkey.IGNORE_REGULAR_EXCLUDES)
+            with_regular_query = solution['query'].intersection(with_regular_query)
+            if with_regular_query:
+                msg = _('All matches were excluded by regular filtering for argument')
+            elif with_modular_query:
+                msg = _('All matches were excluded by modular filtering for argument')
```

Comment:
I used regular filtering and modular filtering key words because you can easily found in documentation. The regular filtering also represents excludes from command line and plugins therefore call them configuration would be a little bit confusing.

Reply:
Proposing:
```

    All matches were excluded by regular filtering for argument

    All matches were filtered out by modular filtering for argument
```

Reply:
@j-mracek what about:
```
All matches were filtered out by exclude filtering for argument
All matches were filtered out by modular filtering for argument
```
The modular phrase is the same as yours, the "regular" (still disliking the word) one uses the "exclude filtering" wording, which I think well describes the filtering is of "excludes", which can happen in all those various places. For the argument of being able to google the formulation to find the right documentation, I think a google query like "exclude filtering dnf" will be unique enough and work well. I hope you like it :slightly_smiling_face: